unstructured/ingest/interfaces.py

```diff
@@ -654,22 +655,54 @@ def __init__(self, write_config: WriteConfig, connector_config: BaseConnectorCon
         self.write_config = write_config
         self.connector_config = connector_config
 
+    def conform_dict(self, data: dict) -> None:
+        """
+        When the original dictionary needs to be modified in place
+        """
+        return
+
+    def normalize_dict(self, element_dict: dict) -> dict:
+        """
+        When the original dictionary needs to be mapped to a new one
+        """
+        return element_dict
+
     @abstractmethod
     def initialize(self):
         """Initializes the connector. Should also validate the connector is properly
         configured."""
 
-    @abstractmethod
     def write(self, docs: t.List[BaseSingleIngestDoc]) -> None:
-        pass
+        elements_dict = self.get_elements_dict(docs=docs)
+        self.modify_and_write_dict(elements_dict=elements_dict)
+
+    def get_elements_dict(self, docs: t.List[BaseSingleIngestDoc]) -> t.List[t.Dict[str, t.Any]]:
+        dict_list: t.List[t.Dict[str, t.Any]] = []
+        for doc in docs:
+            local_path = doc._output_filename
+            with open(local_path) as json_file:
+                dict_content = json.load(json_file)
+                logger.info(
+                    f"Extending {len(dict_content)} json elements from content in {local_path}",
+                )
+                dict_list.extend(dict_content)
+        return dict_list
 
     @abstractmethod
     def write_dict(self, *args, elements_dict: t.List[t.Dict[str, t.Any]], **kwargs) -> None:
         pass
 
+    def modify_and_write_dict(
+        self, *args, elements_dict: t.List[t.Dict[str, t.Any]], **kwargs
+    ) -> None:
```

Comment:
nit: maybe just a quick line docstring... modify=conform+normalize and why it's needed.